### PR TITLE
Minor tweaks

### PR DESCRIPTION
**Pull request by ltratt**
A few not-very-related minor changes: each commit is self contained.

### CODE REVIEW

File: ykrt/src/location.rs [hunk 1]
```diff
@@ -100,36 +100,36 @@ impl Location {
     pub fn new() -> Self {
         // Locations start in the counting state with a count of 0.
         Self {
-            state: AtomicUsize::new(State::new().x),
+            state: AtomicUsize::new(LocationInternalState::new().x),
```
Comment (vext01) [inline ykrt/src/location.rs:103]: Would `LocationInner` be more idiomatic of this repo? It's a bit shorter too :)

**Comment by vext01** [comment]
Just one query. The rest LGTM.

File: ykrt/src/location.rs (hunk 1 above)
Comment (ltratt) [reply ykrt/src/location.rs]: Good idea. Fixed in https://github.com/ykjit/yk/pull/459/commits/5dd09b2be515f7c30386523b1a7c5fbbb3a8a957.

**Comment by vext01** [comment]
Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**